dask_geomodeling/tests/test_geometry.py

```diff
@@ -391,6 +391,92 @@ def test_area_empty(self):
         self.assertEqual(0, len(result))
 
 
+class TestGeometryWktSource(unittest.TestCase):
+    def setUp(self):
+        self.projection = "EPSG:28992"
+        self.geometry = shapely_transform(
+            box(135000.5, 455998, 135001.5, 455999.5),
+            "EPSG:28992", self.projection
+        )
+        self.request = dict(
+            mode='intersects',
+            geometry=box(135000.5, 455998, 135001.5, 455999.5),
+            projection=self.projection,
+        )
+
+    def test_geometry_wkt_source_vals_filters_no_supported(self):
+        view = geometry.GeometryWKTSource(self.geometry.wkt, self.projection)
+        self.request['filters'] = dict(name='test')
+        with self.assertRaises(ValueError) as ctx:
+            view.get_data(**self.request)
+        self.assertEqual("Filter are not supported", str(ctx.exception))
+
+    def test_geometry_wkt_source_vals_wrong_mode(self):
+        self.request['mode'] = 'jose'
+        view = geometry.GeometryWKTSource(self.geometry.wkt, self.projection)
+        with self.assertRaises(ValueError) as ctx:
+            view.get_data(**self.request)
+        self.assertEqual("Unknown mode 'jose'", str(ctx.exception))
+
+    def test_geometry_wkt_source_vals(self):
+        self.request['mode'] = 'intersects'
+        view = geometry.GeometryWKTSource(self.geometry.wkt, self.projection)
+        actual = view.get_data(**self.request)
+        assert actual["features"]['geometry'][0].wkt == self.geometry.wkt
+
+    def test_geometry_wkt_source_vals_empty(self):
+        self.request['mode'] = 'intersects'
+        self.geometry = shapely_transform(
+            box(135100.5, 455998, 135101.5, 455999.5),
+            "EPSG:28992", self.projection
+        )
+        view = geometry.GeometryWKTSource(self.geometry.wkt, self.projection)
+        actual = view.get_data(**self.request)
+        assert actual["features"].empty
+
+    def test_geometry_wkt_source_vals_mode_centroid(self):
+        self.request['mode'] = 'centroid'
+        view = geometry.GeometryWKTSource(self.geometry.wkt, self.projection)
+        actual = view.get_data(**self.request)
+        assert actual["features"]['geometry'][0].wkt == self.geometry.wkt
+
+    def test_geometry_wkt_source_vals_centroid_empty(self):
+        self.request['mode'] = 'centroid'
+        self.geometry = shapely_transform(
+            box(135100.5, 455998, 135101.5, 455999.5),
```

Comment:
Can you make this such that the geometry is actually intersecting the request geometry, but its centroid is not?